packages/product-react/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,34 @@
+import clsx from 'clsx';
+import React from 'react';
+
+export interface ICheckboxProps extends React.HTMLProps<HTMLInputElement> {
+  className?: string;
+  id?: string;
+  label?: string;
+  name?: string;
+}
+
+export const Checkbox: React.FC<ICheckboxProps> = ({
+  className,
+  id,
+  name,
+  label,
+  ...rest
+}) => {
+  return (
+    <div className={clsx('ray-checkbox', className)}>
+      <input
+        className="ray-checkbox__input"
+        type="checkbox"
+        id={id}
+        name={name}
+        {...rest}
+      />
+      <label className="ray-checkbox__label" htmlFor={id}>
```

Comment:
i wonder if you should make the id required here to prevent that htmlFor from being `undefined`. You could also  you can nest the `<input>` directly inside the `<label>` and the htmlFor and id attributes are not needed because the association is implicit
```
<label><input/>{label}</label>
```

Reply:
ya good point for the required id , the way the may the made it in the ray CSS ,the focus stop working if you don't put the htmlFor as an attribute. really weird :/